From 1d549bb837594094201cd1c40d457737aec8b5a1 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Tue, 2 Dec 2014 18:21:53 -0500
Subject: [fs] NFS: Don't try to reclaim delegation open state if recovery failed

Message-id: <1417544515-7381-4-git-send-email-steved@redhat.com>
Patchwork-id: 101083
O-Subject: [RHEL7.1 PATCH 3/5] NFS: Don't try to reclaim delegation open state if recovery failed
Bugzilla: 1166845
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

If state recovery failed, then we should not attempt to reclaim delegated
state.

http://lkml.kernel.org/r/CAN-5tyHwG=Cn2Q9KsHWadewjpTTy_K26ee+UnSvHvG4192p-Xw@mail.gmail.com
Cc: stable@vger.kernel.org
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: f8ebf7a8ca35dde321f0cd385fee6f1950609367
Bugzilla: 1166845
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/delegation.c b/fs/nfs/delegation.c
index 5853f53..4904a1d 100644
--- a/fs/nfs/delegation.c
+++ b/fs/nfs/delegation.c
@@ -125,6 +125,8 @@ again:
    continue;
   if (!test_bit(NFS_DELEGATED_STATE, &state->flags))
    continue;
+  if (!nfs4_valid_open_stateid(state))
+   continue;
   if (!nfs4_stateid_match(&state->stateid, stateid))
    continue;
   get_nfs_open_context(ctx);
-- 
1.7.1